src/HDF5.jl

```diff
@@ -20,7 +20,7 @@ iscontiguous, ishdf5, ismmappable, name,
 o_copy, o_delete, o_open, p_create,
 readmmap, @read, @write, root, set_dims!, t_create, t_commit
 ## Public Types:
-# Attribute, File, Group, Dataset, Datatype, Opaque
+# H5DataType, Attribute, File, Group, Dataset, Datatype, Opaque
```

Comment:
I'm not crazy on `H5DataType` being the name because it sounds too close to `Datatype`. What about `H5DataStore`?

Reply:
Yeah that sounds better. Are we ok with this having the `H5` prefix, which I think is appropriate since it's the supertype? In your opinion `H5DataStore` is better than `H5DataFile` (which seems not as accurate)

Reply:
Yeah, I'm good with the H5 prefix since users which are subtyping on it are like to have done an `import`, so being explicit in its name seems fine/good to me. I don't have a strong opinion on `H5DataFile` vs `H5DataStore`, though the fact that `Group <: H5Data(Store|File)` makes me think `File` isn't the ideal suffix.